src/platform/bouffalolab/BL602/wifi_mgmr_portable.c

```diff
@@ -103,3 +111,43 @@ int wifi_mgmr_profile_ssid_get(uint8_t * ssid)
 
     return profile_msg.ssid_len;
 }
+
+uint32_t wifi_mgmr_get_security_type(void)
+{
+    /** return values defined from app::Clusters::WiFiNetworkDiagnostics::WiFiVersionEnum */
+    if (strlen(wifiMgmr.wifi_mgmr_stat_info.passphr))
+    {
+        if (gWpaSm.pmk_len)
+        {
+            if (WPA_PROTO_WPA == gWpaSm.proto)
+            {
+                return 3;
+            }
+            else if (WPA_PROTO_RSN == gWpaSm.proto)
+            {
+                if (gWpaSm.key_mgmt & (WPA_KEY_MGMT_PSK | WPA_KEY_MGMT_PSK_SHA256))
+                {
+                    return 4;
+                }
+                else
+                {
+                    return 5;
+                }
+            }
+            else if (WPA_PROTO_WAPI == gWpaSm.proto)
+            {
+                return 6;
```

Comment:
6 is not a valid value of SecurityTypeEnum.

Reply:
Removed WPA_PROTO_WAPI.